packages/flutter_driver/lib/src/extension/extension.dart

```diff
@@ -372,12 +404,30 @@ class FlutterDriverExtension {
   }
 
   Finder _createFinder(SerializableFinder finder) {
-    final FinderConstructor constructor = _finders[finder.finderType];
-
-    if (constructor == null)
-      throw 'Unsupported finder type: ${finder.finderType}';
-
-    return constructor(finder);
+    switch (finder.finderType) {
+      case 'ByText':
+        return _createByTextFinder(finder as ByText);
+      case 'ByTooltipMessage':
+        return _createByTooltipMessageFinder(finder as ByTooltipMessage);
+      case 'BySemanticsLabel':
+        return _createBySemanticsLabelFinder(finder as BySemanticsLabel);
+      case 'ByValueKey':
+        return _createByValueKeyFinder(finder as ByValueKey);
+      case 'ByType':
+        return _createByTypeFinder(finder as ByType);
+      case 'PageBack':
+        return _createPageBackFinder();
+      case 'Ancestor':
+        return _createAncestorFinder(finder as Ancestor);
+      case 'Descendant':
+        return _createDescendantFinder(finder as Descendant);
+      default:
+        if (_finderExtensions.containsKey(finder.finderType)) {
+          return _finderExtensions[finder.finderType].createFinder(finder);
+        }else{
```

Comment:
```suggestion
        } else {
```

Reply:
Also, up to you, but you could just drop the `else` entirely here since you're returning in your if.